components/BattleScreen/Menu/Menu.js

```diff
@@ -1,28 +1,122 @@
 import styled from "styled-components";
+import { useState } from "react";
 
 export default function Menu({ onAttack, disabled }) {
+	const [showAttackMenu, setShowAttackMenu] = useState(false);
+
 	const handleAttackClick = () => {
 		if (!disabled) {
```

Comment:
What exactly is not disabled here? It would be better to use positive naming: `isDisplayed` or similar; by doing so, you don't have to double negate before you know what's happening.